CMSgov/mint-app
# Easi 2365/op needs set needed

# EASI-2365

## Changes and Description

- Possible Operational Need table updated with the needed logic to say if a need is needed or not
    - Associated Task List Section is exposed through GQL for Possible Operational Need an Operational Need table
- Operational Need trigger
    -  This uses the new meta data added to the possible operational need data when specific columns are changed. If they are changed to match the condition specified, the need will be updated with either needed / not needed
    - If a column different from the one that controls the logic for that need is set, the operational need will not be updated.  
    - If there is an array of columns that specify a need, the need will be set as needed if at least 1 of the columns is set in a way that satisfies the condition  see `Process participant appeals` operational need



## How to test this change

1. Create a model Plan
2. Update task list sections based on the mapping in the possible operational need table.
3. Verify that the need gets toggled to needed / not needed based on the logic.
## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
